.changelog/25998.txt

```diff
@@ -0,0 +1,11 @@
+```release-note:security
+template: Add the insecure sprig_htpasswd and sprig_derive sprig functions to the DefaultTemplateFunctionDenylist.
```

Comment:
These are the right function names for the items we've added to the denylist. But also, why are these being denied? What's insecure about them? The original problem that the advisory was about was for functions that allowed recursive operations and blew up memory consumption. Why aren't they in the list here?

Also, this is the user-facing changelog, so the configuration we refer to should be the user facing [`function_denylist`](https://developer.hashicorp.com/nomad/docs/job-specification/template#function_denylist)

Reply:
"Insecure" is conditional on use, and in this case it's no different than having a `{{ secret }}` block. They're just warning you not to write secrets to disk, and for Nomad that applies the same which is why we give people `template.env` or the `NOMAD_SECRETS_DIR`